extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -32,34 +47,53 @@ public void setNodeName(final TransactionManagerConfiguration transactions) {
         }
     }
 
-    public void setDefaultProperties(Properties properties) {
-        //TODO: this is a huge hack to avoid loading XML parsers
-        //this needs a proper SPI
-        try {
-            Field field = PropertiesFactory.class.getDeclaredField("delegatePropertiesFactory");
-            field.setAccessible(true);
-            field.set(null, new QuarkusPropertiesFactory(properties));
-
-        } catch (Exception e) {
-            log.error("Could not override transaction properties factory", e);
-        }
-
-        defaultProperties = properties;
-    }
-
     public void setDefaultTimeout(TransactionManagerConfiguration transactions) {
         transactions.defaultTransactionTimeout.ifPresent(defaultTimeout -> {
             arjPropertyManager.getCoordinatorEnvironmentBean().setDefaultTimeout((int) defaultTimeout.getSeconds());
             TxControl.setDefaultTimeout((int) defaultTimeout.getSeconds());
         });
     }
 
-    public static Properties getDefaultProperties() {
-        return defaultProperties;
-    }
-
     public void disableTransactionStatusManager() {
         arjPropertyManager.getCoordinatorEnvironmentBean()
                 .setTransactionStatusManagerEnable(false);
     }
+
+    public void setConfig(TransactionManagerConfiguration transactions, NarayanaQuarkusConfiguration config) {
```

Comment:
would not be better to unite the `TransactionManagerConfiguration` and `NarayanaQuarkusConfiguration` under one class. It would be easier to get where the configuration is defined

Reply:
+1